ykbh/src/lib.rs

```diff
@@ -16,9 +18,109 @@ pub struct StackFrame {
 }
 
 impl StackFrame {
+    /// Free the allocated memory this stack frame points to.
     fn free(&mut self) {
         unsafe { dealloc(self.locals, self.layout) }
     }
+
+    /// Given a pointer `src` and a size, write its value to the pointer `dst`.
+    pub fn write_val(&mut self, dst: *mut u8, src: *const u8, size: usize) {
+        unsafe {
+            std::ptr::copy(src, dst, size);
+        }
+    }
+    /// Write a constant to the pointer `dst`.
+    pub fn write_const(&mut self, dest: *mut u8, constant: &Constant) {
```

Comment:
Changing the way we represent consts has now moved up my priorities.

With a bag of bytes and a type tag, we'd never need to destructure like this: just copy the bytes.

 We'd probably get a little speedup for it.

See #139

Reply:
Yeah, I'm not a fan of the current representation either.